### PR TITLE
Refactor interactive pose viewer backend to 3Dmol

### PR DESCRIPTION
**Pull request by JenkeScheen**
We're switching to a `3DMol` backend instead of `NGL`. Beside having a more '`PyMol`-like' atom/bonds representation and controls, this package is _much more flexible_ and will allow us to:
- show interactions in a modular way 
- show bond orders from SDF

Overall the code should become somewhat cleaner/more interpretable. 

Main points to implement:
- [x] use `airium` to write HTML rather than just slapping strings together
- [x] add view orientation for all targets (possibly remove 272)
- [x] feed in color_dict for both `subpockets` and `fitness` coloring methods (so remove bfactor workaround)
- [x] add interactions by default. may need to use `PLIP` for this because `ProLIF` seems to sensitive to protonation variations
- [x] finalize bond orders and protonation representation for both ligand and protein
- [x] make sure all components work in the workflow
- [x] remove all redundant code 

Preview of new view:
image

### CODE REVIEW

**Comment by dotsdl** [comment]
@JenkeScheen is this ready for review, or is there more to hit on your checklist first?

**Comment by JenkeScheen** [comment]
Thanks, this is still WIP. I'd like to get the bypass in place, then get a review in before I attempt to remove the current flow.

**Comment by dotsdl** [comment]
Sounds good! Ping me when you'd like a review!

**Comment by chrisiacovella** [comment]
Have you considered using templates for the html generation part?   I am a big fan of [jinja](https://jinja.palletsprojects.com/en/3.1.x/ ), since, in addition to just straight up find/replace, it allows you to define formatting macros and loops within the templates, minimizing the amount of processing/preparation you need to do inside the python code.

**Comment by JenkeScheen** [comment]
noting down here so I don't forget to update the conda env.yml: we're adding a dependency to `PLIP`, of which the `openbabel` dependency is a bit fiddly. This ended up doing the trick:
```
conda install -c conda-forge openbabel lxml 
conda install -c conda-forge --no-deps plip
```

**Comment by JenkeScheen** [comment]
`PLIP` conda install has `abi` issues on running `my_mol.analyze()`:
```
libc++abi: terminating with uncaught exception of type swig::stop_iteration
Abort trap: 6
```
for now, going with `singularity` -> load `PLIP` docker image approach.

**Comment by JenkeScheen** [comment]
This is almost ready for review, I just need to resolve some issues with protonation. I'll check in with @kaminow for this, looks like it's mostly just `OE` being difficult. 

@dotsdl, I'm mostly unsure about https://github.com/choderalab/asapdiscovery/pull/443/commits/73cfad4187c1b5c9368ca88b64d1230098369a1d. It works on my end, but I'm not sure that `conda`-installing `plip` will result in a `plip` bin for all users, if not then https://github.com/choderalab/asapdiscovery/commit/ecdf23db1443406e48a22a6be44bf45363a0ec4f?diff=unified#diff-8ad06bf1f92be6cc4e927ac68d370da58af3abfef7c2fa0e9c4ceac4a12c8756R278 would fail. Can you check that after install, that CLI call works correctly?

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (dotsdl) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:355]: Think I agree with @chrisiacovella that using `jinja` plus a template may be less painful than building up the script below. I think you could also use a combination if you prefer, using `airium` to build up the overall HTML doc but using `jinja` to populate the JS script.

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (dotsdl) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:294]: Tests currently choke on this: looks like `intn_dict_xml["report"]["bindingsite"]` is a `list`, with more than one item. Not sure what the correct thing to do here is; do we expect more than one binding site in many cases?

**Review by dotsdl — CHANGES_REQUESTED** [review]
Looking improved over previous version @JenkeScheen! A few changes needed before this is good to merge:

- CI currently failing. Looks like you'll need to remove some of the test items since this PR removes things like `view_coords_MERS_CoV_Mpro_7ene`. Also, there is a failure in testing `get_interactions_plip`: it looks like there may be more than one bindingsite in many cases?

In a future PR, we'll probably want to:
- use `jinja` templates for at least the JS components of the generated HTML artifacts; this is likely to be simpler than building up the strings hardcoded into the module itself.
- move target specs in `viz_targets`, and `view_coords`, `pocket_dict`s, and `color_dict`s in `_gif_blocks`, and `orient`s in `_html_blocks` to a set of YAML files instead of directly in the module code. We can take advantage of similar patterns @hmacdope has done elsewhere in `asapdiscovery`

**Comment by dotsdl** [comment]
Running the tests also leaves a bunch of log files in the working directory; can we disable these?

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (JenkeScheen) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:358]: I agree. I'll discuss with @hmacdope and @mikemhenry before starting a PR to introduce `jinja` throughout the codebase.

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (JenkeScheen) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:297]: ah, that might be happening if we have two ligands bound to a dimer (in case of e..g Mpro). In principle that's fine, we can show interactions on both monomers. I'll change this to account for that.

**Comment by JenkeScheen** [comment]
> CI currently failing. Looks like you'll need to remove some of the test items since this PR removes things like view_coords_MERS_CoV_Mpro_7ene

ah, I'll have a look. I haven't been paying attention to CI as it was breaking because of an issue with `fragalysis`, thanks for spotting.

**Comment by dotsdl** [comment]
@JenkeScheen looks like CI is broken in a different spot now in `get_interactions_plip`. Can you address this?

**Comment by JenkeScheen** [comment]
ugh

**Comment by JenkeScheen** [comment]
@dotsdl looks like CI still fails after my bugfixes, but on something to do with `fragalysis`.

**Comment by JenkeScheen** [comment]
looks like CI is still failing stochastically on `fragalysis` stuff. `python 3.10` succeeded this time, so this PR is good to go. @dotsdl did you want to have one last look?

**Comment by hmacdope** [comment]
> looks like CI is still failing stochastically on `fragalysis` stuff. `python 3.10` succeeded this time, so this PR is good to go. @dotsdl did you want to have one last look?

Yeah sorry about this known issue #463

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (dotsdl) [inline asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py:304]: Is the intention here to just overwrite the dict with the values of the last `'interactions'` dict in `binding_site`? Is this guaranteed to always be what we want to do?

**Review by dotsdl — APPROVED** [review]
Still seeing lots of `log.txt` files polluting the working directory when running the tests. Will raise an issue for this; library code generally shouldn't be creating side-effect files in the working directory by default.

Otherwise, good to merge once comments addressed.

File: asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py
Comment (JenkeScheen) [reply asapdiscovery-dataviz/asapdiscovery/dataviz/html_viz.py]: keys here are unique, we're enumerating interactions when we get them from PLIP. so in this way, no interaction will be overwritten at any point.

**Pull request merged**
